src/foam/approval/ApprovableAwareDAO.js

```diff
@@ -0,0 +1,450 @@
+      sendSingleRequest(x, request, request.getInitiatingUser());
+      approverIds.remove(request.getInitiatingUser());
+
+      for ( int i = 0; i < approverIds.size(); i++ ) {
+        sendSingleRequest(getX(), request, approverIds.get(i));
+      }
+      `
+    },
+    {
+      name: 'remove_',
+      javaCode: `
+        User user = (User) x.get("user");
+        Logger logger = (Logger) x.get("logger");
+
+        // system and admins override the approval process
+        if ( user != null && ( user.getId() == User.SYSTEM_USER_ID || user.getGroup().equals("admin") || user.getGroup().equals("system") ) ) return super.remove_(x,obj);
+
+        ApprovableAware approvableAwareObj = (ApprovableAware) obj;
+        DAO approvalRequestDAO = ((DAO) getX().get("approvalRequestDAO"));
+
+        List approvedObjRemoveRequests = ((ArraySink) approvalRequestDAO
+          .where(
+            foam.mlang.MLang.AND(
+              foam.mlang.MLang.EQ(ApprovalRequest.DAO_KEY, getDaoKey()),
+              foam.mlang.MLang.EQ(ApprovalRequest.OBJ_ID, approvableAwareObj.getApprovableKey()),
+              foam.mlang.MLang.EQ(RoleApprovalRequest.OPERATION, Operations.REMOVE),
+              foam.mlang.MLang.EQ(RoleApprovalRequest.IS_FULFILLED, false),
+              foam.mlang.MLang.OR(
+                foam.mlang.MLang.EQ(ApprovalRequest.STATUS, ApprovalStatus.APPROVED),
+                foam.mlang.MLang.EQ(ApprovalRequest.STATUS, ApprovalStatus.REJECTED)
+              )
+            )
+          ).select(new ArraySink())).getArray();
+
+        if ( approvedObjRemoveRequests.size() == 1 ) {
+          RoleApprovalRequest fulfilledRequest = (RoleApprovalRequest) approvedObjRemoveRequests.get(0);
+          fulfilledRequest.setIsFulfilled(true);
+
+          approvalRequestDAO.put_(getX(), fulfilledRequest);
+
+          if ( fulfilledRequest.getStatus() == ApprovalStatus.APPROVED ) {
+            return super.put_(x,obj);
+          } 
+          
+          return null;  // as request has been REJECTED
+        } 
+        
+        if ( approvedObjRemoveRequests.size() > 1 ) {
+          logger.error("Something went wrong cannot have multiple approved/rejected requests for the same request!");
+          throw new RuntimeException("Something went wrong cannot have multiple approved/rejected requests for the same request!");
+        } 
+
+        RoleApprovalRequest approvalRequest = new RoleApprovalRequest.Builder(getX())
+          .setDaoKey(getDaoKey())
+          .setObjId(approvableAwareObj.getApprovableKey())
+          .setClassification(getOf().getObjClass().getSimpleName())
+          .setOperation(Operations.REMOVE)
+          .setInitiatingUser(((User) x.get("user")).getId())
+          .setStatus(ApprovalStatus.REQUESTED).build(); 
+
+        fullSend(getX(), approvalRequest, obj);
+
+        // we need to prevent the remove_ call from being passed all the way down to actual deletion since
```

Comment:
remove commented code .. add it back when you have the proper implementation (save it in a JIRA card or something until then)

Reply:
Removed remove_ method cuz will never be used under approvable